templates/environment/versions/cf-v1.4.1.yml

```diff
@@ -0,0 +1,393 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+Description: CloudFormation environment template for infrastructure shared among Copilot workloads.
+Metadata:
+  Version: 'v1.4.1'
+Parameters:
+  AppName:
+    Type: String
+  EnvironmentName:
+    Type: String
+  ALBWorkloads:
+    Type: String
+    Default: ""
+  EFSWorkloads:
+    Type: String
+    Default: ""
+  NATWorkloads:
+    Type: String
+    Default: ""
+  ToolsAccountPrincipalARN:
+    Type: String
+  AppDNSName:
+    Type: String
+    Default: ""
+  AppDNSDelegationRole:
+    Type: String
+    Default: ""
+  Aliases:
+    Type: String
+    Default: ""
+Conditions:
+  CreateALB:
+    !Not [!Equals [ !Ref ALBWorkloads, "" ]]
+  DelegateDNS:
+    !Not [!Equals [ !Ref AppDNSName, "" ]]
+  ExportHTTPSListener: !And
```

Comment:
Does this condition evaluate to true for imported certs?

Reply:
actually i don't think we need this condition for `ImportCertificate` resources at all since we just need to depend on the https listener. The hookup will only be created when creating the HTTPS listener and import cert arns are specified.

Reply:
OK but will the HTTPSListener be created if they don't have an `AppDNSName`? If I'm understanding right that doesn't seem to be the case.

So we won't be able to attach the certificates right? or am I missing something

Reply:
addressed! see the manual test results in https://github.com/aws/copilot-cli/pull/2413#issuecomment-853246123